lib/inventory_sync/async/inventory_full_sync.rb

```diff
@@ -0,0 +1,97 @@
+require 'rest-client'
+
+module InventorySync
+  module Async
+    class InventoryFullSync < ::ApplicationJob
+      def perform(organization)
+        @organization = organization
+        @subscribed_hosts_ids = Set.new(
+          ForemanInventoryUpload::Generators::Queries.for_slice(Host).pluck(:id)
```

Comment:
If you are limiting yourself to a single org, you should filter only the relevant hosts:
```suggestion
          ForemanInventoryUpload::Generators::Queries.for_slice(Host.unscoped.where(organization: organization)).pluck(:id)
```